### DISCUSSION TITLE
ERROR: cannot package project VRoidRetargetAttachActor.h(11): Error: Enum 'EAttachType' shares engine name 'EAttachType' with enum 'EAttachType'

### DISCUSSION THREAD

**Discussion by ToffelsKater** [pixiv/vroid-sdk-developers#168] [Q&A] [answered]
I get packaging errors for UE 5.3.2 VROID SDK 1.0
VRoidRetargetAttachActor.h(11): Error: Enum 'EAttachType' shares engine name 'EAttachType' with enum 'EAttachType'
VRoidRetargetAttachActor.h(19): Error: Struct 'FVrmSpawnParameter' shares engine name 'VrmSpawnParameter' with struct 'FVrmSpawnParameter'
Error: Struct 'FVrmSpawnParameter' shares engine name 'VrmSpawnParameter' with struct 'FVrmSpawnParameter'

**Comment by mameoPix** [comment]
Thank you for your feedback!
We have confirmed that the same problem occurs on our end.

Please try the following workaround
Try changing the **LoadingPhase** of **VRoidSdkCore** in VRoidSdk.uplugin to `PreDefault`.
```
{
	"Name": "VRoidSdkCore",.
	"Type": "Runtime",.
	"LoadingPhase": "PreDefault",.
	"PlatformAllowList": [
		"Win64".
	],.
},.
```
If LoadingPhase is Default, the FVrmSpawnParameter in VRoidDataTypes.h is treated as undefined at packaging time, which may be the cause.

**Reply by mameoPix** [reply]
Sorry, we were providing a fix code for a different error content here.
We have not yet been able to reproduce the error you mentioned.
We will continue to check the situation.

**Reply by ToffelsKater** [reply]
It does not work for me, i still get the same error message

**Reply by mameoPix** [reply]
Could you please tell me the minimum configuration that reproduces the error in this case?
1. Create a Third Person Character Template project.
2. Added VRoid SDK and VRM4U to the project, connected AttachRetargetCharacter node after BeginPlay of ThirdPersonCharacter and did Compile.
3. Could not reproduce it at hand by creating a package in this state.

The environment is VRoidSDKforUE1.0.0 and VRM4U:20240728, UE5.3.2 obtained via launcher.
![image](https://github.com/user-attachments/assets/e28386be-2f95-4ffb-984f-7793a22560a7)

**Reply by ToffelsKater** [reply]
What package configuration did you use?
did you tried Release build(no developer build or debug build)

**Reply by mameoPix** [reply]
Build Configuration was checked with Development.
Switching to Shippnig did not confirm the error.
Do you have any idea what else could be causing this?
![image](https://github.com/user-attachments/assets/e9b720b7-3c6a-4a34-81b1-05908b4889ee)

**Reply by ToffelsKater** [reply]
I have no idea, I might just try reinstalling the SDK and see if that works

**Reply by mameoPix** [reply, marked as answer]
Thank you for your patience.
The name of EAttachType may be changed in the future as it may conflict with the engine code in the future.

Also, since it cannot be reproduced in a newly created project
A possible solution would be to delete the temporary files Binaries, Saved, Intermediate, etc.
However, unfortunately, we do not know if these solutions will work well.
